common/src/stack/command/stack_completions.sh

```diff
@@ -0,0 +1,31 @@
+_stack_completion() {
+	local loc cur opts BASE_DIR
+
+	# BASE_DIR represents the path to the directory containing
+	# all the possible commands in their directories.
+	BASE_DIR="./stack/commands/"
+
+	# Currently typed word
+	cur="${COMP_WORDS[COMP_CWORD]}"
+
+	# Generate directory path from the input command line,
+	# excluding the first word (stack)
+	loc=${BASE_DIR}"$(tr -s ' ' '/' <<< ${COMP_WORDS[@]:1})"
+
+	# Make sure that loc is actually path to a directory
+	if [[ ! -d ${loc} ]]; then
+		if [[ -n ${cur} ]]; then
+			loc=$(echo ${loc} | rev | cut -d/ -f2- | rev)
+		else
+			return
+		fi
+	fi
+	# Generate options, depending on loc
+	opts=$(find ${loc} -maxdepth 1 -mindepth 1 -type d -exec basename '{}' \;)
```

Comment:
The bug I recalled probably comes from this line.  It occurs when the code attempts to match a parameter after an `=`.

stack list pallet expanded=[TAB]

```
sd-stacki-213:/export/stack # stack list pallet expanded=find: ‘/opt/stack/lib/python3.*/site-packages/stack/commands/list/pallet/expanded’: No such file or directory
find: ‘/opt/stack/lib/python3.*/site-packages/stack/commands/list/pallet/expanded’: No such file or directory

carts/        drivers/      isos/         pallets/      sbin/         spreadsheets/
```

Note that `expanded` is a stacki parameter for the `list pallet` command, which your code does not currently have a convenient way to access.  It expects a boolean value, but it would be reasonable to suppress the `find` error and keep the listing of the current working directory (sometimes a parameter is `file=some/file`).

Reply:
I did suppress the errors. Please give it a look!